### PR TITLE
Unbox primitives when marshalling to id

### PR DESCRIPTION
**Pull request by fealebenpae**

### CODE REVIEW

File: src/NativeScript/ObjC/ObjCTypes.mm [hunk 1]
```diff
@@ -118,7 +119,7 @@ id toObject(ExecState* execState, const JSValue& value) {
     }
 
     if (value.isString()) {
-        return [NSString stringWithString:(NSString*)value.toString(execState)->value(execState).createCFString().get()];
+        return [static_cast<NSString *>(jsCast<JSString*>(value)->value(execState)) copy];
```
Comment (jasssonpet) [inline src/NativeScript/ObjC/ObjCTypes.mm]: I wonder if we should `autorelease` the copy.

Comment (fealebenpae) [inline src/NativeScript/ObjC/ObjCTypes.mm]: Yes, we should, and I'll fix it right away.

**Comment by jasssonpet** [comment]
:+1:

I wonder if it's worth adding the same logic when we marshall to types other than `id`.

**Comment by fealebenpae** [comment]
We don't need to. Calling `JSValue::toNumber(ExecState*)` performs unboxing for us. Same with strings and booleans.

**Pull request merged**